src/hdmf/term_set.py

```diff
@@ -0,0 +1,97 @@
+from collections import namedtuple
+from .utils import docval
+
+
+class TermSet():
+    """
+    Class for implementing term sets from ontologies and other resources used to define the
+    meaning and/or identify of terms.
+
+    :ivar name: The name of the TermSet
+    :ivar term_schema_path: The LinkML YAML enumeration schema
+    :ivar ontology_version: The version of the ontology
+    :ivar ontology_source_name:  The name of the ontology
+    :ivar ontology_source_uri: The uri of the ontology
+    """
+    def __init__(self,
+                 term_schema_path: str,
+                 ):
+        """
+        :param term_schema_path: The path to LinkML YAML enumeration schema
+
+        """
+        try:
+            from linkml_runtime.utils.schemaview import SchemaView
+        except ImportError:
+            msg = "Install linkml_runtime"
+            raise ValueError(msg)
+        self.term_schema_path = term_schema_path
+        self.view = SchemaView(self.term_schema_path)
+        self.sources = self.view.schema.prefixes
+
+    def __repr__(self):
+        re = "class: %s\n" % str(self.__class__)
+        re += "term_schema_path: %s\n" % self.term_schema_path
+        return re
+
+    def _perm_value_key_info(self, perm_values_dict: dict, key: str):
+        """
```

Comment:
Private functions should have a double `__` as prefix. A single underscore is "protected" in Python.